### DISCUSSION TITLE
vpn client on iphone 15 disconnect issue

### DISCUSSION THREAD

**Discussion by rc4u** [strongswan/strongswan#2308] [Q&A]
We see an issue on vpn client running on iphone 15 that the connect is dropping when the device go offline and not established again when the device go online again.
The conf on the server is:
conn ios
       fragmentation=yes
       keyexchange=ikev1
       authby=xauthrsasig
       xauth=server
       left=%defaultroute
       leftsubnet=0.0.0.0/0
       leftfirewall=yes
       leftcert=serverCert.pem
       right=%any
       rightsourceip=10.0.0.0/16
       auto=add
       reauth=no
       
     and the log on the server on the client session:
     Jun 24 06:35:19 15[CFG] <ios|4227>   using certificate "CN=1-DEVICEID, O=VENDORID"
Jun 24 06:35:19 15[CFG] <ios|4227>   using trusted ca certificate "C=CN, O=VENDORID, CN=VENDORID CA"
Jun 24 06:35:19 15[CFG] <ios|4227> checking certificate status of "CN=1-DEVICEID, O=VENDORID"
Jun 24 06:35:19 15[CFG] <ios|4227> certificate status is not available
Jun 24 06:35:19 15[CFG] <ios|4227>   reached self-signed root ca with a path length of 0
Jun 24 06:35:19 15[IKE] <ios|4227> authentication of 'CN=1-DEVICEID, O=VENDORID' with RSA_EMSA_PKCS1_NULL successful
Jun 24 06:35:19 15[IKE] <ios|4227> authentication of 'CN=*.VENDORID.com' (myself) successful
Jun 24 06:35:19 15[IKE] <ios|4227> sending end entity cert "CN=*.VENDORID.com"
Jun 24 06:35:19 15[IKE] <ios|4227> sending issuer cert "C=US, ST=Arizona, L=Scottsdale, O=Starfield Technologies, Inc., OU=http://certs.starfieldtech.com/repository/, CN=Starfield Secure Certificate Authority - G2"
Jun 24 06:35:19 15[ENC] <ios|4227> generating ID_PROT response 0 [ ID CERT CERT SIG ]
Jun 24 06:35:19 15[ENC] <ios|4227> splitting IKE message (3356 bytes) into 3 fragments
Jun 24 06:35:19 15[ENC] <ios|4227> generating ID_PROT response 0 [ FRAG(1) ]
Jun 24 06:35:19 15[ENC] <ios|4227> generating ID_PROT response 0 [ FRAG(2) ]
Jun 24 06:35:19 15[ENC] <ios|4227> generating ID_PROT response 0 [ FRAG(3/3) ]
Jun 24 06:35:19 15[NET] <ios|4227> sending packet: from 185.241.4.126[4500] to 147.235.207.143[7828] (1248 bytes)
Jun 24 06:35:19 15[NET] <ios|4227> sending packet: from 185.241.4.126[4500] to 147.235.207.143[7828] (1248 bytes)
Jun 24 06:35:19 15[NET] <ios|4227> sending packet: from 185.241.4.126[4500] to 147.235.207.143[7828] (968 bytes)
Jun 24 06:35:19 15[ENC] <ios|4227> generating TRANSACTION request 2542856642 [ HASH CPRQ(X_USER X_PWD) ]
Jun 24 06:35:19 15[NET] <ios|4227> sending packet: from 185.241.4.126[4500] to 147.235.207.143[7828] (92 bytes)
Jun 24 06:35:19 05[NET] <ios|4227> received packet: from 147.235.207.143[7828] to 185.241.4.126[4500] (92 bytes)
Jun 24 06:35:19 05[ENC] <ios|4227> parsed TRANSACTION response 2542856642 [ HASH CPRP(X_USER X_PWD) ]
Jun 24 06:35:19 05[IKE] <ios|4227> XAuth authentication of 'user' successful
Jun 24 06:35:19 05[ENC] <ios|4227> generating TRANSACTION request 2675535272 [ HASH CPS(X_STATUS) ]
Jun 24 06:35:19 05[NET] <ios|4227> sending packet: from 185.241.4.126[4500] to 147.235.207.143[7828] (92 bytes)
Jun 24 06:35:19 11[NET] <ios|4227> received packet: from 147.235.207.143[7828] to 185.241.4.126[4500] (92 bytes)
Jun 24 06:35:19 11[ENC] <ios|4227> parsed TRANSACTION response 2675535272 [ HASH CPA(X_STATUS) ]
Jun 24 06:35:19 11[IKE] <ios|4227> IKE_SA ios[4227] established between 185.241.4.126[CN=*.VENDORID.com]...147.235.207.143[CN=1-DEVICEID, O=VENDORID]
Jun 24 06:35:19 11[IKE] <ios|4227> scheduling rekeying in 10058s
Jun 24 06:35:19 11[IKE] <ios|4227> maximum IKE_SA lifetime 10598s
Jun 24 06:35:19 12[NET] <ios|4227> received packet: from 147.235.207.143[7828] to 185.241.4.126[4500] (188 bytes)
Jun 24 06:35:19 12[ENC] <ios|4227> unknown attribute type (28683)
Jun 24 06:35:19 12[ENC] <ios|4227> parsed TRANSACTION request 2022268918 [ HASH CPRQ(ADDR MASK DNS NBNS EXP VER U_BANNER U_DEFDOM U_SPLITDNS U_SPLITINC U_LOCALLAN U_PFS U_SAVEPWD U_FWTYPE U_BKPSRV (28683)) ]
Jun 24 06:35:19 12[IKE] <ios|4227> peer requested virtual IP %any
Jun 24 06:35:19 12[CFG] <ios|4227> reassigning offline lease to 'user'
Jun 24 06:35:19 12[IKE] <ios|4227> assigning virtual IP 10.0.0.27 to peer 'user'
Jun 24 06:35:19 12[ENC] <ios|4227> generating TRANSACTION response 2022268918 [ HASH CPRP(ADDR DNS) ]
Jun 24 06:35:19 12[NET] <ios|4227> sending packet: from 185.241.4.126[4500] to 147.235.207.143[7828] (92 bytes)
Jun 24 06:35:19 13[NET] <ios|4227> received packet: from 147.235.207.143[7828] to 185.241.4.126[4500] (380 bytes)
Jun 24 06:35:19 13[ENC] <ios|4227> parsed QUICK_MODE request 763869082 [ HASH SA No ID ID ]
Jun 24 06:35:19 13[CFG] <ios|4227> selected proposal: ESP:AES_CBC_256/HMAC_SHA2_256_128/NO_EXT_SEQ
Jun 24 06:35:19 13[ENC] <ios|4227> generating QUICK_MODE response 763869082 [ HASH SA No ID ID ]
Jun 24 06:35:19 13[NET] <ios|4227> sending packet: from 185.241.4.126[4500] to 147.235.207.143[7828] (188 bytes)
Jun 24 06:35:19 01[NET] <ios|4227> received packet: from 147.235.207.143[7828] to 185.241.4.126[4500] (76 bytes)
Jun 24 06:35:19 01[ENC] <ios|4227> parsed QUICK_MODE request 763869082 [ HASH ]
Jun 24 06:35:19 01[IKE] <ios|4227> CHILD_SA ios{3219} established with SPIs c0a55631_i 051b7812_o and TS 0.0.0.0/0 === 10.0.0.27/32
Jun 24 06:35:41 08[NET] <ios|4227> received packet: from 147.235.207.143[7828] to 185.241.4.126[4500] (92 bytes)
Jun 24 06:35:41 08[ENC] <ios|4227> parsed INFORMATIONAL_V1 request 969971267 [ HASH D ]
Jun 24 06:35:41 08[IKE] <ios|4227> received DELETE for ESP CHILD_SA with SPI 051b7812
Jun 24 06:35:41 08[IKE] <ios|4227> closing CHILD_SA ios{3219} with SPIs c0a55631_i (65165 bytes) 051b7812_o (51427 bytes) and TS 0.0.0.0/0 === 10.0.0.27/32
Jun 24 06:35:41 09[NET] <ios|4227> received packet: from 147.235.207.143[7828] to 185.241.4.126[4500] (108 bytes)
Jun 24 06:35:41 09[ENC] <ios|4227> parsed INFORMATIONAL_V1 request 3355018742 [ HASH D ]
Jun 24 06:35:41 09[IKE] <ios|4227> received DELETE for IKE_SA ios[4227]
Jun 24 06:35:41 09[IKE] <ios|4227> deleting IKE_SA ios[4227] between 185.241.4.126[CN=*.VENDORID.com]...147.235.207.143[CN=1-DEVICEID, O=VENDORID]
Jun 24 06:35:41 09[CFG] <ios|4227> lease 10.0.0.27 by 'user' went offline

any idea why the connect is not reconnecting on the client (iphone) ?

**Comment by rc4u** [comment]
it look like for this connection there is a definition of IKE_SA lifetime, but I cannot find where it is defined ? and why for some device it is activate and for other it is not...
Jun 24 06:35:19 11[IKE] <ios|4227> scheduling rekeying in 10058s
Jun 24 06:35:19 11[IKE] <ios|4227> maximum IKE_SA lifetime 10598s

**Reply by tobiasbrunner** [reply]
> it look like for this connection there is a definition of IKE_SA lifetime, but I cannot find where it is defined ?

There are defaults. Read the man page/docs.

**Comment by tobiasbrunner** [comment]
You know that iOS supports IKEv2? Why would you use IKEv1?
